contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -198,23 +169,57 @@ func (t *T) Has(f FlagType) (exists bool) {
 	return
 }
 
-func New(ctx api.Context, id string, pod *api.Pod) (*T, error) {
+func (t *T) Roles() []string {
+	var roles []string
+
+	if r, ok := t.Pod.ObjectMeta.Labels[annotation.RolesKey]; ok {
+		roles = strings.Split(r, ",")
```

Comment:
more concretely the `not(emptyRole)` filters `roleA,,,,,,roleB` to `roleA,roleB`. I'll introduce a `trimmed` predicate, to make the `string.TrimSpace` to be also part of the filter.